pages/plant/[id].js

```diff
@@ -0,0 +1,121 @@
+import Image from "next/image";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import BackLink from "@/components/BackLink";
+
+const StyledSeasonList = styled.ul`
+  list-style: none;
+  padding: 0;
+`;
+
+const IconsContainer = styled.article`
+  display: flex;
+  justify-content: space-around;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const ImageBorder = styled.article`
+  position: relative;
+  margin-top: 20px;
+  height: 55vw;
+  width: 90vw;
+  border: 0px solid;
+  border-radius: 15px;
+  overflow: hidden;
+  margin: auto;
+`;
+
+export default function PlantDetails({ plants }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const plantData = plants ? plants.find((plant) => plant.id === id) : null;
+
```

Comment:
maybe you could also check if the router isReady: 
```suggestion
 if (!router.isReady) return <div>Loading...</div>;
```